db/log_format.h

```diff
@@ -17,7 +17,7 @@
 namespace ROCKSDB_NAMESPACE {
 namespace log {
 
-enum RecordType {
+enum RecordType : int {
   // Zero is reserved for preallocated files
```

Comment:
### Temporary note to reviewer: Please ignore changes to db/log_format.h. It belongs to https://github.com/facebook/rocksdb/pull/13225 this PR has to base onto.